### ISSUE TITLE
Aggregation over indexed list is out of order and deduplicated

### ISSUE THREAD

**Issue by MattFerraro** [elastic/elasticsearch#12547]
It looks like the actual document data indexed for the field "nums" is a list of 4 floats in descending order, but when the aggregation is run to collect the data, only 3 floats are returned and they are in ascending order.

```bash
#! /bin/sh

# Delete the index
curl -XDELETE "http://localhost:9200/test"

# Insert a document with one field -- a list of floats
curl -XPUT "http://localhost:9200/test/doc/1?refresh=true" -d '{
  "nums": [5.0, 3.5, 3.5, 1.0]
}'

# Expected:
# aggregations.foobar.value == [5.0, 3.5, 3.5, 1.0]

# Actual:
# aggregations.foobar.value == [1.0, 3.5, 5.0]
curl -XGET "http://localhost:9200/test/_search" -d '{
    "query": {
        "match_all": {}
    },
    "filter": {},
    "aggregations": {
        "foobar": {
            "scripted_metric": {
                "init_script": "_agg[\"numList\"] = []",
                "map_script": "if (doc != null) { _agg.numList.add(doc[\"nums\"].getValues()) }",
                "combine_script": "return _agg.numList",
                "reduce_script": "return _aggs.flatten()"
            }
        }
    }
}'

```

**Comment by MattFerraro** [comment]
I've also opened this up for discussion on [stackoverflow](http://stackoverflow.com/questions/31712933/elasticsearch-aggregation-over-indexed-list-is-out-of-order-and-deduplicated)

**Comment by jasontedor** [comment]
This is due to the way that [multivalued fields](https://www.elastic.co/guide/en/elasticsearch/guide/current/complex-core-fields.html#_multivalue_fields) are indexed. Since you are attempting to access the input values (including duplicates), you'll want to use the `_source` field in your aggregation.

**Comment by MattFerraro** [comment]
Ah I see, thanks for the info.

**Comment by clintongormley** [comment]
@MattFerraro @jasontedor using `_source` for aggregations will be very very very slow. I think you'll have to do something like using a separate field for each number (lousy solution I know).

An alternative would be to index ["val1_5.0", "val2_3.5", "val3_3.5", etc]. Another poor solution.  

Not sure if your script is representative of what you really want to do, but you could use a `nested` doc for each value, and use a `nested` aggregation to access them.

**Comment by MattFerraro** [comment]
Yep, after trying with _source I found that the data appeared as expected, but the aggregation took so long that it became unusable. I suppose that makes a lot of sense because we're accessing non-indexed data.

We went ahead and indexed the exact floats we wanted as nested fields in a separate spot and now access them as ```doc['nums.num1']``` and ```docs['nums.num2']```, and it is both quick and correct!
